### PR TITLE
Jw simulator update

### PR DESCRIPTION
**Pull request by Vuwij**

### CODE REVIEW

File: flatland_server/include/flatland_server/layer.h [hunk 1]
```diff
@@ -69,6 +69,8 @@ class Layer : public Entity {
   Body *body_ = nullptr;
   CollisionFilterRegistry *cfr_;  ///< collision filter registry
   std::string viz_name_;          ///< for visualization
+  std::string map_path_;
```
Comment (hendrik-skubch) [inline flatland_server/include/flatland_server/layer.h]: What is this used for?

File: flatland_server/include/flatland_server/layer.h [hunk 2]
```diff
@@ -69,6 +69,8 @@ class Layer : public Entity {
...
+  double resolution_;
```
Comment (hendrik-skubch) [inline flatland_server/include/flatland_server/layer.h]: What is resolution exactly? A double seems to be an odd choice

Comment (Vuwij) [reply flatland_server/include/flatland_server/layer.h]: 0.05, the scale of the screen compared to the map. Each pixel is 0.05 meters. (apparently its called resolution in the code so I'll just use that

File: flatland_server/include/flatland_server/layer.h (hunk 1 above)
Comment (Vuwij) [reply flatland_server/include/flatland_server/layer.h]: Keeps the map path in the layer, so the human navigation plugin can find the map path and generate the costmap/ path planning

Comment (hendrik-skubch) [reply flatland_server/include/flatland_server/layer.h]: Alright. In this case, i suggest to group both members in a struct (e.g., MapDescription) and make it a const member, as it is not supposed to change during the lifetime of a layer.
e.g., const MapDescription map_info_;

File: flatland_server/src/layer.cpp [hunk 3]
```diff
@@ -64,10 +64,12 @@ namespace flatland_server {
...
 Layer::Layer(b2World *physics_world, CollisionFilterRegistry *cfr,
              const std::vector<std::string> &names, const Color &color,
-             const Pose &origin, const cv::Mat &bitmap, double occupied_thresh,
-             double resolution, const YAML::Node &properties)
+             const Pose &origin, const std::string &map_path,
+             const cv::Mat &bitmap, double occupied_thresh, double resolution,
+             const YAML::Node &properties)
     : Entity(physics_world, names[0]),
       names_(names),
+      map_description_{map_path, static_cast<float>(resolution)},
```
Comment (hendrik-skubch) [inline flatland_server/src/layer.cpp]: why do you cast to float here?

Comment (Vuwij) [reply flatland_server/src/layer.cpp]: double to float

File: flatland_server/include/flatland_server/layer.h [hunk 4]
```diff
@@ -70,6 +70,16 @@ class Layer : public Entity {
   CollisionFilterRegistry *cfr_;  ///< collision filter registry
   std::string viz_name_;          ///< for visualization
 
+  struct MapDescription {
+    std::string path_;
+    double resolution_;
+    MapDescription() {}
+    MapDescription(std::string path, float resolution)
```
Comment (hendrik-skubch) [inline flatland_server/include/flatland_server/layer.h]: Why is this a float?

**Review by hendrik-skubch — APPROVED** [review]

**Pull request merged**